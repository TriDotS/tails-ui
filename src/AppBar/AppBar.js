import {
  AppBar,
  Close,
  DesktopMenu,
  DesktopMenuItem,
  Hamburger,
  MenuItemIcon,
  MobileHeaderText,
  MobileMenu,
  MobileMenuContainer,
  MobileMenuHeader,
  MobileMenuItem,
  Title,
} from './styles'
import React, { Component } from 'react'

export default class extends Component {
  state = {
    showMenu: false,
  }

  toggleMenu = () => {
    this.setState({
      showMenu: !this.state.showMenu,
    })
  }

  render() {
    const {
      title,
      menuItems,
      backgroundColor,
      fontColor,
      mobileMenuOpenIcon,
      mobileMenuCloseIcon,
      shadowed,
      fixed,
      height,
      mobileMenuHeaderColor,
      mobileMenuHeaderTextColor,
      mobileMenuHeaderText,
    } = this.props

    return (
      <AppBar
        backgroundColor={backgroundColor}
        fontColor={fontColor}
        shadowed={shadowed}
        fixed={fixed}
        height={height}
      >
        <Title fontColor={fontColor}>{title}</Title>
        <DesktopMenu fontColor={fontColor}>
          {menuItems.map((link, index) => {
            return (
              <span key={index}>
                {link.click
                  ? <DesktopMenuItem fontColor={fontColor} onClick={link.click}>
                      {link.text}
                    </DesktopMenuItem>
                  : <DesktopMenuItem fontColor={fontColor}>
                      {link.link}
                    </DesktopMenuItem>}
              </span>
            )
          })}
        </DesktopMenu>
        <Hamburger onClick={this.toggleMenu}>
          {mobileMenuOpenIcon || 'menu'}
        </Hamburger>
        <MobileMenu
          showMenu={this.state.showMenu}
          style={{ display: this.state.showMenu ? 'flex' : 'none' }}
        >
          <MobileMenuHeader mobileMenuHeaderColor={mobileMenuHeaderColor}>
            <MobileHeaderText
              mobileMenuHeaderTextColor={mobileMenuHeaderTextColor}
            >
              {mobileMenuHeaderText}
            </MobileHeaderText>
            <Close onClick={this.toggleMenu}>
              {mobileMenuCloseIcon || 'close'}
            </Close>
          </MobileMenuHeader>
          <MobileMenuContainer>
            {menuItems.map((link, index) => {
              return (
                <span key={index}>
                  <MobileMenuItem key={index} onClick={this.toggleMenu}>
                    <MenuItemIcon>{link.icon && link.icon}</MenuItemIcon>
                    {link.link}
                  </MobileMenuItem>
                </span>
              )
            })}
          </MobileMenuContainer>
        </MobileMenu>
      </AppBar>
    )
  }
}
