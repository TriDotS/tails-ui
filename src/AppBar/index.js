import {
  AppBar,
  Close,
  DesktopMenu,
  DesktopMenuItem,
  Hamburger,
  MobileMenu,
  MobileMenuItem,
  Title,
} from './styles'
import React, { Component } from 'react'

export default class extends Component {
  state = {
    showMenu: false,
  }

  toggleMenu = () => {
    this.setState({
      showMenu: !this.state.showMenu,
    })
  }

  render() {
    const {
      title,
      menuItems,
      backgroundColor,
      fontColor,
      mobileMenuOpenIcon,
      mobileMenuCloseIcon,
      hasShadow,
      fixed,
      height,
    } = this.props

    return (
      <AppBar
        backgroundColor={backgroundColor}
        fontColor={fontColor}
        hasShadow={hasShadow}
        fixed={fixed}
        height={height}
      >
        <Title fontColor={fontColor}>{title}</Title>
        <DesktopMenu fontColor={fontColor}>
          {menuItems.map((link, index) => {
            return (
              <span key={index}>
                {link.click
                  ? <DesktopMenuItem fontColor={fontColor} onClick={link.click}>
                      {link.text}
                    </DesktopMenuItem>
                  : <DesktopMenuItem fontColor={fontColor}>
                      {link.link}
                    </DesktopMenuItem>}
              </span>
            )
          })}
        </DesktopMenu>
        <Hamburger onClick={this.toggleMenu}>
          {mobileMenuOpenIcon || 'menu'}
        </Hamburger>
        <MobileMenu
          showMenu={this.state.showMenu}
          style={{ display: this.state.showMenu ? 'flex' : 'none' }}
        >
          <Close onClick={this.toggleMenu}>
            {mobileMenuCloseIcon || 'close'}
          </Close>
          {menuItems.map((link, index) => {
            return (
              <span key={index}>
                {link.click
                  ? <MobileMenuItem onClick={link.click}>
                      {link.text}
                    </MobileMenuItem>
                  : <MobileMenuItem key={index} onClick={this.toggleMenu}>
                      {link.link}
                    </MobileMenuItem>}
              </span>
            )
          })}
        </MobileMenu>
      </AppBar>
    )
  }
}
