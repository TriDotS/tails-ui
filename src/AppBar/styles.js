import styled, { keyframes } from 'styled-components'

// Slide in animation for mobile menu
const slideIn = keyframes`
  from {
	  margin-left: 100%;
    width: 100%;
	}

	to {
	  margin-left: 0%;
    width: 100%;
	}
`

// Main app bar component
export const AppBar = styled.div`
  display: flex;
  align-items: center;
	width: 100%;
  background: ${props => props.backgroundColor || 'royalblue'};
  color: ${props => props.fontColor || 'black'};
  height: ${props => props.height || '3em'};
  box-shadow: ${props => (props.hasShadow ? '1px 1px 1px #999' : null)};
  position: ${props => (props.fixed ? 'fixed' : 'initial')};
`

// Title Component
export const Title = styled.div`
  font-size: 1.5em;
  color: ${props => props.fontColor || 'black'};
  padding-left: 0.5em;
`

// Hamburger icon
export const Hamburger = styled.div`
  padding-right: 0.5em;
  font-size: 2em;
  margin-left: auto;
  cursor: pointer;

  @media (min-width: 700px) {
		display: none;
	}
`

// Close icon for mobile menu
export const Close = styled.div`
  position: absolute;
  padding-right: 15px;
  padding-top: 5px;
  top: 0;
  right: 0;

  @media(min-width: 700px) {
    display: none;
  }
`

// Main mobile menu panel
export const MobileMenu = styled.div`
  height: 100vh;
  background-color: rgba(255, 255, 255, 0.95);
  color: black;
  z-index: 100;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 2em;
  text-align: center;
  animation: 0.3s alternate ${slideIn};

  @media (min-width: 700px) {
		display: none;
	}
`

// Mobile meu items
export const MobileMenuItem = styled.div`
  text-transform: uppercase;
  padding-bottom: 15px;
  color: black;

  > a {
    text-decoration: none;
    color: black;
  }
`

export const DesktopMenu = styled.div`
  font-weight: bolder;
  margin-left: auto;
  color: ${props => props.fontColor || 'black'};

  @media (max-width: 700px) {
		display: none;
	}
`

export const DesktopMenuItem = styled.span`
  padding-right: 15px;
  cursor: pointer;
  color: ${props => props.fontColor || 'black'};

  > a {
    text-decoration: none;
    color: ${props => props.fontColor || 'black'};
  }
`
