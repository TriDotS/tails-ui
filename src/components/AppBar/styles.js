import slideFromRight from '../../animations/slideFromRight'
import styled from 'styled-components'

// Main app bar component
export const AppBar = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  background: ${props => props.backgroundColor || 'royalblue'};
  color: ${props => props.fontColor || 'black'};
  height: ${props => props.height || '3em'};
  box-shadow: ${props => (props.shadowed ? '1px 1px 1px #999' : null)};
  position: ${props => (props.fixed ? 'fixed' : 'initial')};
`

// Title Component
export const Title = styled.div`
  font-size: 1.5em;
  color: ${props => props.fontColor || 'black'};
  padding-left: 0.5em;
`

// Hamburger icon
export const Hamburger = styled.div`
  padding-right: 0.5em;
  font-size: 2em;
  margin-left: auto;
  cursor: pointer;

  @media (min-width: 700px) {
    display: none;
  }
`

// Close icon for mobile menu
export const Close = styled.div`
  margin-left: auto;
  padding-right: 10px;

  @media (min-width: 700px) {
    display: none;
  }
`

// Main mobile menu panel
export const MobileMenu = styled.div`
  height: 100vh;
  background: #fafafa;
  color: black;
  z-index: 10000000;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  display: flex;
  flex-direction: column;
  animation: 0.3s alternate ${slideFromRight};

  @media (min-width: 700px) {
    display: none;
  }
`

export const MobileMenuHeader = styled.div`
  background: ${props => props.mobileMenuHeaderColor || 'royalblue'};
  height: 3em;
  top: 0;
  width: 100%;
  display: flex;
  align-items: center;
`

export const MobileHeaderText = styled.div`
  color: ${props => props.mobileMenuHeaderTextColor || 'white'};
  padding-left: 10px;
  font-size: 1.5em;
`

// Mobile meu items
export const MobileMenuItem = styled.div`
  text-transform: uppercase;
  padding-bottom: 10px;
  padding-top: 10px;
  border-bottom: 1px solid #eee;
  color: #424242;
  font-size: 1.1em;

  > a {
    text-decoration: none;
    color: #424242;
    font-size: 1.1em;
  }
`

export const RemoveDecoration = styled.span`
  text-transform: uppercase;
  padding-bottom: 10px;
  padding-top: 10px;
  border-bottom: 1px solid #eee;
  color: #424242;
  font-size: 1.1em;

  > a {
    text-decoration: none;
    color: #424242;
    font-size: 1.1em;
  }
`

export const DesktopMenu = styled.div`
  font-weight: bolder;
  margin-left: auto;
  color: ${props => props.fontColor || 'black'};

  @media (max-width: 700px) {
    display: none;
  }
`

export const DesktopMenuItem = styled.span`
  padding-right: 15px;
  cursor: pointer;
  color: ${props => props.fontColor || 'black'};

  > a {
    text-decoration: none;
    color: ${props => props.fontColor || 'black'};
  }
`

export const MenuItemIcon = styled.span`margin-right: 10px;`

export const MobileMenuContainer = styled.div`
  background: white;
  margin: 20px;
  text-align: center;
`
