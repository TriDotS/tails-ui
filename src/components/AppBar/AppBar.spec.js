import AppBar from './AppBar'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <AppBar
      menuItems={[
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
      ]}
    />,
    div
  )
})

it('generate AppBar snapshot', () => {
  const component = renderer.create(
    <AppBar
      menuItems={[
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
        { text: 'Home', link: <a href="/">Home</a> },
      ]}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
