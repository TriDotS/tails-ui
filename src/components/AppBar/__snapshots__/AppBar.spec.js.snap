// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate AppBar snapshot 1`] = `
<div
  className="sc-bdVaJa fXnJXs"
  height={undefined}
>
  <div
    className="sc-bwzfXH bwNFHM"
  />
  <div
    className="sc-dnqmqq bQbZXj"
  >
    <span>
      <span
        className="sc-iwsKbI efEKy"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
    </span>
    <span>
      <span
        className="sc-iwsKbI efEKy"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
    </span>
    <span>
      <span
        className="sc-iwsKbI efEKy"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
    </span>
    <span>
      <span
        className="sc-iwsKbI efEKy"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
    </span>
  </div>
  <div
    className="sc-htpNat cpMKuo"
    onClick={[Function]}
  >
    menu
  </div>
  <div
    className="sc-ifAKCX fVixIb"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="sc-EHOje fzWXcg"
    >
      <div
        className="sc-bZQynM hJBfGn"
      />
      <div
        className="sc-bxivhb jRVxcX"
        onClick={[Function]}
      >
        close
      </div>
    </div>
    <div
      className="sc-gqjmRU eAwNAW"
    >
      <div
        className="sc-gzVnrw cpPDdq"
      >
        <span
          className="sc-htoDjs nxrqN"
          onClick={[Function]}
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </div>
      <div
        className="sc-gzVnrw cpPDdq"
      >
        <span
          className="sc-htoDjs nxrqN"
          onClick={[Function]}
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </div>
      <div
        className="sc-gzVnrw cpPDdq"
      >
        <span
          className="sc-htoDjs nxrqN"
          onClick={[Function]}
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </div>
      <div
        className="sc-gzVnrw cpPDdq"
      >
        <span
          className="sc-htoDjs nxrqN"
          onClick={[Function]}
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
